base/abstractarraymath.jl

```diff
@@ -411,12 +411,26 @@ end
     eachrow(A::AbstractVecOrMat)
 
 Create a generator that iterates over the first dimension of vector or matrix `A`,
-returning the rows as views.
+returning the row vectors as views.
 
 See also [`eachcol`](@ref) and [`eachslice`](@ref).
 
 !!! compat "Julia 1.1"
      This function requires at least Julia 1.1.
+
+# Example
+
+```jldoctest
+julia> a = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> collect(eachrow(a))
```

Comment:
Likewise, I wonder if it wouldn't be clearer here to print out `first(eachrow(a))`, which will make more obvious the type of what gets returned.

Reply:
Or show both?

Reply:
Thanks, did that. Individual views print nicer, `SubArray` may be a confusing implementation detail.